app.py

```diff
@@ -0,0 +1,137 @@
+import os
+import json
+import requests
+from flask import Flask, redirect, request
+from typing import Dict, Optional
+from id3c.cli.redcap import is_complete
+
+
+REDCAP_API_TOKEN = os.environ['REDCAP_API_TOKEN']
+REDCAP_API_URL = os.environ['REDCAP_API_URL']
+app = Flask(__name__)
+
+
+def fetch_user_data(net_id: str) -> Optional[Dict[str, str]]:
+    """
+    Exports a REDCap record matching the given *net_id*. Returns None if no
+    match is found.
+
+    Raises an :class:`AssertionError` if REDCap returns multiple matches for the
+    given *net_id*.
+    """
+    filter_logic = f'[netid] = "{net_id}"'
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'csvDelimiter': '',
+        'fields[0]': 'netid',
+        'fields[1]': 'record_id',
+        'fields[2]': 'eligibility_screening_complete',
+        'fields[3]': 'consent_form_complete',
+        'fields[4]': 'enrollment_questionnaire_complete',
```

Comment:
Does REDCap understand this form field name syntax?

Per HTTP spec, multiple values for a form field are represented in the POST by a repeated form field name. `requests` does that when you pass an iterable value, e.g.

```
'fields': ['netid', 'record_id', ...],
```

becomes

```
fields=netid&fields=record_id&fields=...
```

in the POST.

But I think REDCap requires a single form value that is a comma-separated list of field names. Here's what we do in ID3C:

https://github.com/seattleflu/id3c/blob/f26086d153990e26df7508bdc63d223a3b20f55e/lib/id3c/cli/redcap.py#L192-L193

Reply:
I actually took the form field name syntax directly from REDCap's API playground.

I can update it to the comma-separated string syntax like we use in ID3C.

Reply:
Addressed in a new commit.

Reply:
Ahh, ok, so REDCap does support that! Good to know. Thanks!